sdk/containers/azcontainerregistry/blob_client_test.go

```diff
@@ -8,99 +8,221 @@ package azcontainerregistry
+		require.NoError(t, err)
+		res, err := client.CheckBlobExists(ctx, repository, blobDigest, nil)
+		require.NoError(t, err)
+		require.Equal(t, blobDigest, *res.DockerContentDigest)
+	})
+
+	t.Run("CheckChunkExists", func(t *testing.T) {
+		startRecording(t)
+		endpoint, cred, options := getEndpointCredAndClientOptions(t)
+		client, err := NewBlobClient(endpoint, cred, &BlobClientOptions{ClientOptions: options})
+		require.NoError(t, err)
+		res, err := client.CheckChunkExists(ctx, repository, blobDigest, "bytes=0-299", nil)
+		require.NoError(t, err)
+		require.NotEmpty(t, *res.ContentLength)
+	})
+
+	t.Run("CheckChunkExists_fail", func(t *testing.T) {
+		startRecording(t)
+		endpoint, cred, options := getEndpointCredAndClientOptions(t)
+		client, err := NewBlobClient(endpoint, cred, &BlobClientOptions{ClientOptions: options})
+		require.NoError(t, err)
+		_, err = client.CheckChunkExists(ctx, repository, "wrong digest", "bytes=0-299", nil)
+		require.Error(t, err)
+	})
+
+	t.Run("CheckBlobExists_fail", func(t *testing.T) {
+		startRecording(t)
+		endpoint, cred, options := getEndpointCredAndClientOptions(t)
+		client, err := NewBlobClient(endpoint, cred, &BlobClientOptions{ClientOptions: options})
+		require.NoError(t, err)
+		_, err = client.CheckBlobExists(ctx, repository, "wrong digest", nil)
+		require.Error(t, err)
+	})
+
+	t.Run("CompleteUpload_wrongDigest", func(t *testing.T) {
+		startRecording(t)
+		endpoint, cred, options := getEndpointCredAndClientOptions(t)
+		client, err := NewBlobClient(endpoint, cred, &BlobClientOptions{ClientOptions: options})
+		require.NoError(t, err)
+		getRes, err := client.GetBlob(ctx, repository, blobDigest, nil)
+		require.NoError(t, err)
+		blob, err := io.ReadAll(getRes.BlobData)
+		require.NoError(t, err)
+		startRes, err := client.StartUpload(ctx, "hello-world", nil)
+		require.NoError(t, err)
+		uploadResp, err := client.uploadChunk(ctx, *startRes.Location, streaming.NopCloser(bytes.NewReader(blob)), nil)
+		require.NoError(t, err)
+		_, err = client.completeUpload(ctx, "sha256:00000000", *uploadResp.Location, nil)
+		require.Error(t, err)
+	})
+
+	t.Run("DeleteBlob_fail", func(t *testing.T) {
+		startRecording(t)
+		endpoint, cred, options := getEndpointCredAndClientOptions(t)
+		client, err := NewBlobClient(endpoint, cred, &BlobClientOptions{ClientOptions: options})
+		require.NoError(t, err)
+		_, err = client.DeleteBlob(ctx, repository, "wrong digest", nil)
+		require.Error(t, err)
+	})
+
+	t.Run("GetBlob", func(t *testing.T) {
+		startRecording(t)
+		endpoint, cred, options := getEndpointCredAndClientOptions(t)
+		client, err := NewBlobClient(endpoint, cred, &BlobClientOptions{ClientOptions: options})
+		require.NoError(t, err)
+		res, err := client.GetBlob(ctx, repository, blobDigest, nil)
+		require.NoError(t, err)
+		require.NotEmpty(t, *res.ContentLength)
+		if recording.GetRecordMode() != recording.PlaybackMode {
+			reader, err := NewDigestValidationReader(blobDigest, res.BlobData)
```

Comment:
Why doesn't this work in playback mode, out of curiosity? Or are we avoiding trying to save off a huge blob (understandable).

Reply:
Good question. The validation reader compares a digest it computes to the server's. This breaks in playback because I recorded a static fake digest instead of the real thing, to prevent tests in general depending on values that could change. But instead of skipping digest validation In playback, these cases should use the `fakeHash` type I added in this PR, which returns a sum matching the recorded value; I'll make that change.